docs/src/developer/reference/oauth.md

```diff
@@ -0,0 +1,428 @@
+  "clientId": "b9e65569-aa61-462d-915d-94c8d6ef17a7",
+  "clientSecret": "3f6fbd62835859b2bac411b2a2a2a54699ec56504ee32099748de3a762d41a2d"
+}
+```
+
+These credentials have to be stored in a safe place and cannot be recovered if they are lost.
+
+### Get authorization code
+
+When the user wants to use the 3rd party app for the first time, they need to authorize it to access Wire resources on their behalf.
+
+To do so the user is redirected to a login screen for authentication. Once authenticated, the user can either grant or deny the client's access request and the corresponding scope (a list of permissions to give to the 3rd party app).
+
+Example request:
+
+```http
+GET /authorize?
+  scope=read%3Aself%20write%3Aconversation&
+  response_type=code&
+  client_id=b9e65569-aa61-462d-915d-94c8d6ef17a7&
+  redirect_uri=https%3A%2F%2Fclient.example.com&
+  state=foobar
+```
+
+Url encoded query parameters:
+
+| Parameter       | Description                                                                                                                                                                                                                                                                       |
+| --------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `scope`         | Required. The scope of the access request.                                                                                                                                                                                                                                        |
+| `response_type` | Required. Value MUST be set to  `code`.                                                                                                                                                                                                                                           |
+| `client_id`     | Required. The client identifier.                                                                                                                                                                                                                                                  |
+| `redirect_url`  | Required. MUST match the URL that was provided during client registration                                                                                                                                                                                                         |
+| `state`         | Required. An opaque value used by the client to maintain state between the request and callback.</br>The authorization server includes this value when redirecting the user-agent back to the client.</br>The parameter SHOULD be used for preventing cross-site request forgery. |
```

Comment:
if you say "should" (and i think "must" would be bad because we need to support whatever other people are doing here, if it's the standard) there should be an integration-test that it works without.  did i miss that test?

Reply:
In the standard this is recommended, and we actually require it. If this doesn't work, we can still change it later, IMO.